middlewares/authzWorkspaceCookieSet.js

```diff
@@ -0,0 +1,21 @@
+const asyncHandler = require('express-async-handler');
+
+const csrf = require('../lib/csrf');
+const config = require('../lib/config');
+
+module.exports = asyncHandler(async (req, res, next) => {
+    // We should only have arrived here if we passed authn/authz and
+    // are authorized to access res.locals.workspace_id. We will set a
+    // short-lived cookie specific to this workspace_id that will let
+    // us bypass authn/authz in the future. This checking is done by
+    // middlewares/authzWorkspaceCookieCheck.js
+    
+    const workspace_id = res.locals.workspace_id;
+    const cookieName = `pl_authz_workspace_${workspace_id}`;
```

Comment:
Nice - as I started reading this PR I was thinking "what about multiple workspaces open at the same time" but you already thought of that!

Reply:
😄